### PR TITLE
fix(nuxt): add import protection for `nuxt/*` exports

### PR DESCRIPTION
**Pull request by danielroe**
### ❓ Type of change
- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

When adding new kit/schema/config subpath exports, we weren't explicitly checking for them.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by pi0 — APPROVED** [review]
I hope this fixes (reproduction-less) issues but seems a legit addition 💯

File: packages/nuxt/test/import-protection.test.ts [hunk 1]
```diff
@@ -14,6 +14,9 @@ const testsToTriggerOn = [
   ['/root/node_modules/@vue/composition-api', 'components/Component.vue', true],
   ['@vue/composition-api', 'components/Component.vue', true],
   ['@nuxt/kit', 'components/Component.vue', true],
+  ['nuxt/config', 'components/Component.vue', true],
+  ['nuxt/kit', 'components/Component.vue', true],
+  ['nuxt/schema', 'components/Component.vue', true],
```
Comment (pi0) [inline packages/nuxt/test/import-protection.test.ts:19]: Shall we include `nuxt` itself to the list or will it be conflicting with subpaths?

File: packages/nuxt/src/core/plugins/import-protection.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ export const vueAppPatterns = (nuxt: Nuxt) => [
   [/(^|node_modules\/)@vue\/composition-api/],
   ...nuxt.options.modules.filter(m => typeof m === 'string').map((m: any) =>
     [new RegExp(`^${escapeRE(m as string)}$`), 'Importing directly from module entry points is not allowed.']),
-  ...[/(^|node_modules\/)@nuxt\/kit/, /^nitropack/]
+  ...[/(^|node_modules\/)@nuxt\/kit/, /(^|node_modules\/)nuxt\/(config|kit|schema)/, /^nitropack/]
```
Comment (pi0) [inline packages/nuxt/src/core/plugins/import-protection.ts:22]: Shall we include `nuxt` itself to the list or will it be conflicting with subpaths?

Comment (danielroe) [reply packages/nuxt/src/core/plugins/import-protection.ts:22]: it's already above on L17.

**Pull request merged**